### PR TITLE
[iOS] UI bug on iPad when using filters in Discussion

### PR DESCRIPTION
**Pull request by forgotvas**
Related to: [iOS] UI bug on iPad when using filters in Discussion #308
Contains fix for iPad discussions filter and sort sheets

Filter|Sort
--|--
f1|f2

### CODE REVIEW

File: Discussion/Discussion/Presentation/Posts/PostsView.swift [hunk 1]
```diff
@@ -242,13 +230,53 @@ public struct PostsView: View {
...
     
+    private var filterButton: some View {
+        Button(action: {
+            showFilterSheet = true
+        }, label: {
+            CoreAssets.filter.swiftUIImage.renderingMode(.template)
+                .foregroundColor(Theme.Colors.accentXColor)
+            Text(viewModel.filterTitle.localizedValue)
+        })
+        .confirmationDialog(DiscussionLocalization.Posts.Alert.makeSelection,
```
Comment (rnr) [inline Discussion/Discussion/Presentation/Posts/PostsView.swift:244]: please follow style "each parameter on own line"

File: Discussion/Discussion/Presentation/Posts/PostsView.swift [hunk 2]
```diff
@@ -242,13 +230,53 @@ public struct PostsView: View {
...
+                            isPresented: $showFilterSheet,
+                            titleVisibility: isPad ? .automatic : .visible,
+                            actions: {
+            ForEach(viewModel.filterInfos) { info in
+                Button(action: {
```
Comment (rnr) [inline Discussion/Discussion/Presentation/Posts/PostsView.swift:249]: please follow style "each parameter on own line"

File: Discussion/Discussion/Presentation/Posts/PostsView.swift [hunk 3]
```diff
@@ -242,13 +230,53 @@ public struct PostsView: View {
...
+    
+    private var sortButton: some View {
+        Button(action: {
+            showSortSheet = true
+        }, label: {
+            CoreAssets.sort.swiftUIImage.renderingMode(.template)
+                .foregroundColor(Theme.Colors.accentXColor)
+            Text(viewModel.sortTitle.localizedValue)
+        })
+        .confirmationDialog(DiscussionLocalization.Posts.Alert.makeSelection,
```
Comment (rnr) [inline Discussion/Discussion/Presentation/Posts/PostsView.swift:266]: please follow style "each parameter on own line"

File: Discussion/Discussion/Presentation/Posts/PostsView.swift [hunk 4]
```diff
@@ -242,13 +230,53 @@ public struct PostsView: View {
...
+                            isPresented: $showSortSheet,
+                            titleVisibility: isPad ? .automatic : .visible,
+                            actions: {
+            ForEach(viewModel.sortInfos) { info in
+                Button(action: {
```
Comment (rnr) [inline Discussion/Discussion/Presentation/Posts/PostsView.swift:271]: please follow style "each parameter on own line"

**Review by rnr — APPROVED** [review]
LGTM, just small code-styling changes

File: Discussion/Discussion/Presentation/Posts/PostsView.swift (hunk 1 above)
Comment (forgotvas) [reply Discussion/Discussion/Presentation/Posts/PostsView.swift]: done

File: Discussion/Discussion/Presentation/Posts/PostsView.swift (hunk 2 above)
Comment (forgotvas) [reply Discussion/Discussion/Presentation/Posts/PostsView.swift]: done

File: Discussion/Discussion/Presentation/Posts/PostsView.swift (hunk 3 above)
Comment (forgotvas) [reply Discussion/Discussion/Presentation/Posts/PostsView.swift]: done

**Review by rnr — DISMISSED** [review]

File: Discussion/Discussion/Presentation/Posts/PostsView.swift [hunk 5]
```diff
@@ -242,13 +230,70 @@ public struct PostsView: View {
...
-                ActionSheet(title: Text(DiscussionLocalization.Posts.Alert.makeSelection), buttons: viewModel.filterButtons)
-            })
         }
     }
     
+    private var filterButton: some View {
+        Button(
+            action: {
+                showFilterSheet = true
+            },
```
Comment (saeedbashir) [inline Discussion/Discussion/Presentation/Posts/PostsView.swift:240]: Here is extra white space and giving the warning `Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)`.

File: Discussion/Discussion/Presentation/Posts/PostsView.swift [hunk 6]
```diff
@@ -242,13 +230,70 @@ public struct PostsView: View {
...
+        .confirmationDialog(
+            DiscussionLocalization.Posts.Alert.makeSelection,
+            isPresented: $showFilterSheet,
+            titleVisibility: isPad ? .automatic : .visible,
+            actions: {
+                ForEach(viewModel.filterInfos) { info in
+                    Button(
+                        action: {
+                            viewModel.filter(by: info)
+                        },
```
Comment (saeedbashir) [inline Discussion/Discussion/Presentation/Posts/PostsView.swift:257]: Same extra white space warning.

File: Discussion/Discussion/Presentation/Posts/PostsView.swift [hunk 7]
```diff
@@ -242,13 +230,70 @@ public struct PostsView: View {
...
+                }
+            }
+        )
+    }
+    
+    private var sortButton: some View {
+        Button(
+            action: {
+                showSortSheet = true
+            },
```
Comment (saeedbashir) [inline Discussion/Discussion/Presentation/Posts/PostsView.swift:271]: Same extra white space warning.

File: Discussion/Discussion/Presentation/Posts/PostsView.swift [hunk 8]
```diff
@@ -242,13 +230,70 @@ public struct PostsView: View {
...
+        .confirmationDialog(
+            DiscussionLocalization.Posts.Alert.makeSelection,
+            isPresented: $showSortSheet,
+            titleVisibility: isPad ? .automatic : .visible,
+            actions: {
+                ForEach(viewModel.sortInfos) { info in
+                    Button(
+                        action: {
+                            viewModel.sort(by: info)
+                        },
```
Comment (saeedbashir) [inline Discussion/Discussion/Presentation/Posts/PostsView.swift:287]: Same extra white space warning.

File: Discussion/Discussion/Presentation/Posts/PostsView.swift (hunk 5 above)
Comment (forgotvas) [reply Discussion/Discussion/Presentation/Posts/PostsView.swift]: done

File: Discussion/Discussion/Presentation/Posts/PostsView.swift (hunk 6 above)
Comment (forgotvas) [reply Discussion/Discussion/Presentation/Posts/PostsView.swift]: done

File: Discussion/Discussion/Presentation/Posts/PostsView.swift (hunk 7 above)
Comment (forgotvas) [reply Discussion/Discussion/Presentation/Posts/PostsView.swift]: done

File: Discussion/Discussion/Presentation/Posts/PostsView.swift (hunk 8 above)
Comment (forgotvas) [reply Discussion/Discussion/Presentation/Posts/PostsView.swift]: done

**Review by saeedbashir — DISMISSED** [review]
LGTM 👍

**Review by rnr — APPROVED** [review]

**Review by volodymyr-chekyrta — APPROVED** [review]

**Pull request merged**